### PR TITLE
initial code for urbavore

### PR DESCRIPTION
**Pull request by otaiwo70**

### CODE REVIEW

File: backend/game/cards/keep_card.py [hunk 1]
```diff
@@ -0,0 +1,12 @@
+from abc import abstractmethod
+
+from game.cards.card import Card
+
+
+class KeepCard(Card):
+    @abstractmethod
+    def immediate_effect(self, player_that_bought_the_card, other_players):
+        raise NotImplementedError
```
Comment (borisv13) [inline backend/game/cards/keep_card.py]: Shouldn't this method be in the Card class?

File: backend/game/cards/keep_card.py [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+
+    def special_effect(self, player_that_bought_the_card, other_players):
```
Comment (borisv13) [inline backend/game/cards/keep_card.py]: Why is this not abstract?

File: backend/game/cards/keep_cards/victory_point_manipulation_cards/test_urbavore.py [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+def test_urbavore_player_add_card(player):
+    Urbavore().immediate_effect(player, None)
+    assert player.card_on_hand.get("Urbavore")
+
+
+def test_urbavore_costs_4_energy():
+    assert Urbavore().cost == 4
+
+
+def test_urbavore_gain_2_star(player):
```
Comment (borisv13) [inline backend/game/cards/keep_cards/victory_point_manipulation_cards/test_urbavore.py]: gain_2_star_when?

File: backend/game/cards/keep_cards/victory_point_manipulation_cards/test_urbavore.py [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+    player.victory_points = 4
```
Comment (borisv13) [inline backend/game/cards/keep_cards/victory_point_manipulation_cards/test_urbavore.py]: Why?

File: backend/game/cards/keep_cards/victory_point_manipulation_cards/test_urbavore.py [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+    player.location = locations.Locations.TOKYO
+    Urbavore().immediate_effect(player, None)
+    Urbavore().special_effect(player, None)
+    assert player.victory_points == 5
```
Comment (borisv13) [inline backend/game/cards/keep_cards/victory_point_manipulation_cards/test_urbavore.py]: 4+2 = 5???

File: backend/game/cards/keep_cards/victory_point_manipulation_cards/test_urbavore.py (hunk 4 above)
Comment (otaiwo70) [reply backend/game/cards/keep_cards/victory_point_manipulation_cards/test_urbavore.py]: Removed the preset, was looking to better show that it added 1 to 4 as opposed to default of adding 1 to 0

File: backend/game/cards/keep_cards/victory_point_manipulation_cards/test_urbavore.py (hunk 5 above)
Comment (otaiwo70) [reply backend/game/cards/keep_cards/victory_point_manipulation_cards/test_urbavore.py]: My mistake corrected the test case to add 1 when in tokyo

File: backend/game/cards/keep_cards/victory_point_manipulation_cards/test_urbavore.py (hunk 3 above)
Comment (otaiwo70) [reply backend/game/cards/keep_cards/victory_point_manipulation_cards/test_urbavore.py]: added additional clarity to the test case to improve as needed

File: backend/game/cards/keep_card.py (hunk 2 above)
Comment (otaiwo70) [reply backend/game/cards/keep_card.py]: updated to abstract

File: backend/game/cards/keep_cards/victory_point_manipulation_cards/urbavore.py [hunk 6]
```diff
@@ -0,0 +1,20 @@
+from game.cards.card import Card
+from game.cards.keep_card import KeepCard
+from game.values import locations
+
+
+class Urbavore(KeepCard):
+    def __init__(self):
+        Card.__init__(self, "Urbavore", 4,
+                      "gain 1 extra [star] when beginning your turn in Tokyo. If your are in Tokyo and you roll at least one [attack] add [attack] to your roll",
```
Comment (gpfaia) [inline backend/game/cards/keep_cards/victory_point_manipulation_cards/urbavore.py]: This card only is activated both on turn start and turn roll.  Both those conditions need to be met and a function needs to be created to not just add victory points but add attack if rolled.

**Comment by cvennel** [comment]
closes #143

**Comment by gpfaia** [comment]
@cvennel Debating where to put the tests for the resolving of dice, I could put them in with the card or I could put them in with the dice resolver, thoughts?

File: backend/game/cards/keep_cards/victory_point_manipulation_cards/test_urbavore.py [hunk 7]
```diff
@@ -0,0 +1,42 @@
...
+
+def test_urbavore_gains_no_stars_when_not_in_tokyo(player):
+    player.location = locations.Locations.OUTSIDE
+    Urbavore().special_effect(player, None)
+    assert player.victory_points == 0
+
+
+def test_urbavore_adding_to_attack_when_in_tokyo_and_attack_greater_than_one(player):
+    player.location = locations.Locations.TOKYO
+    attack = Urbavore().attack_dice_special_effect(player, 3)
```
Comment (cvennel) [inline backend/game/cards/keep_cards/victory_point_manipulation_cards/test_urbavore.py]: Being nitpicky, but I'd suggest making local variables to explain what the cards actually doing. Like `attack_rolled = 3` then `assert attack == attack_rolled + 1`

**Pull request merged**